index.html

```diff
@@ -317,18 +313,29 @@
                     loading: false
                 },
                 created: function() {
-                    const searchParams = new URLSearchParams(window.location.search);
+                    let searchParams = new URLSearchParams(window.location.search);
                     if (searchParams.has('stream')) {
                         const queryStream = searchParams.get('stream');
-                        if (["stable", "testing", "next", "testing-devel", "next-devel", "bodhi-updates", "developer"].includes(queryStream)) {
+                        if (this.streamList.includes(queryStream)) {
                             this.stream = queryStream;
+                        } else {
+                            searchParams.set('stream', this.stream);
                         }
+                        history.replaceState(null, null, `${window.location.origin}?${searchParams.toString()}`);
                     }
 
                     this.refreshBuilds();
                     },
-                watch: { stream: 'refreshBuilds' },
+                watch: {
+                    stream() {
+                        this.updateURL();
+                        this.refreshBuilds();
+                    }
+                },
                 methods: {
+                    updateURL: function() {
+                        history.pushState(null, null, `${window.location.origin}?stream=${this.stream}`);
```

Comment:
Hmm actually, shouldn't this be `window.location`? Right now, it's swallowing `/browser`.